### DISCUSSION TITLE
How can i use @unocss/transformer-variant-group in @unocss/runtime

### DISCUSSION THREAD

**Discussion by JoshuaLi319** [unocss/unocss#3762] [Q&A]
My code is as follows, but it doesn't work.

`  <script type="module">
    import transformerVariantGroup from "@unocss/transformer-variant-group";

    window.__unocss = {
      runtime: {
        bypassDefined: true,
      },
      // 自定义指令
      shortcuts: {
        "flex-between": "flex justify-between items-center",
        "flex-center": "flex justify-center items-center",
        "flex-around": "flex justify-around items-center",
      },
      transformers: [transformerVariantGroup()],
    };
  </script>
  <script src="/src/lib/unocssRuntime.js"></script> `

**Comment by jdkcoder** [comment]
bump

**Comment by ackvf** [comment]
```
// unoConfig
  transformers: [
    transformerVariantGroup(),
    {
      get name() {
        console.trace("[transformers].name")
        return "[transformers].name"
      },
      transform(code, id, ctx) {
        console.trace("[transformers].transform", code, id, ctx)
      },
      enforce: "pre",
    },
    transformerDirectives(),
  ],
```

This reveals that only `name` is called here

https://github.com/unocss/unocss/blob/217a594683b4997ba0b936ebb135243e936e5294/packages/core/src/config.ts#L230

and that's it. No other console is called, suggesting that **transformers are not invoked at all**.

**Reply by ackvf** [reply]
https://github.com/unocss/unocss/issues/4386

**Comment by ackvf** [comment]
This is how you can call the transformer yourself:

```
// <script src="https://unpkg.com/magic-string/dist/magic-string.umd.js"></script>
// import * as unocss from "@unocss/core"
// import unoConfig from "../uno.config.ts"

const uno = unocss.createGenerator(unoConfig)

const tokens = new Set<string>()

const update = async () => {
  const u = await uno
  const css = (await u.generate(tokens)).css
  const m = new MagicString(css)
  await u.config.transformers[0].transform(m, undefined, { uno: u } as any)
  // styleRef.textContent = m.toString() // inject the result in a document.head style tag
  return m.toString()
}
```
